optimizer_wrapper.rb

```diff
@@ -200,7 +200,8 @@ def self.solve(service_vrp, job = nil, block = nil)
       multi_modal = Interpreters::MultiModal.new(vrp, service)
       optim_result = multi_modal.multimodal_routes
     elsif vrp.vehicles.empty? || (vrp.services.empty? && vrp.shipments.empty?)
-      optim_result = Wrappers::Wrapper.new.empty_result(service.to_s, vrp, 'No vehicle available for this service')
+      optim_result = config[:services][service].empty_result(
+        service.to_s, vrp, 'No vehicle available for this service', false)
```

Comment:
in fact we do not need the `solver` argument -- i.e., `service.to_s` each solver needs to surcharge the empty_result function so that, wrapper gets the necessary information from the child class.

For example, we can have the following function in `ortools.rb`, `vroom.rb` and `demo.rb`
```
def empty_result(*args)
  super("ortools"/"vroom"/"demo", *args)
end
```

We can leave this for another time as well. But we should keep in mind that solves (ortools,vroom,demo) are based on `Wrapper` class so, they can overload its functions for this kind of purposes and call the original funciton

Reply:
I didn't test the code above by the way but the implementation is along the lines...